documentation/src/main/asciidoc/public/reference/_mapping-indexedembedded.adoc

```diff
@@ -236,6 +236,60 @@ When Hibernate Search is unable to handle an association,
 it will throw an exception on bootstrap.
 If this happens, refer to <<mapping-reindexing-basics>> to know more.
 
+[[mapping-indexedembedded-includeEmbeddedObjectId]]
+== Embedding the entity identifier
+
+Mapping a property as an <<mapping-identifiermapping,identifier>> in the indexed-embedded type
+will not automatically result into it being embedded when using `@IndexedEmbedded` on that type,
+because document identifiers are not fields.
+
+To embed the data of such a property, you can use `@IndexedEmbedded(includeEmbeddedObjectId = true)`,
+which will have Hibernate Search automatically insert a field in the resulting embedded object
+for the indexed-embedded type's <<mapping-identifiermapping-basics,identifier property>>.
+
+The index field will be defined as if the following <<mapping-directfieldmapping,field annotation>>
+was put on the identifier property of the embedded type:
+`@GenericField(searchable = Searchable.YES, projectable = Projectable.YES)`.
+The name of the index field will be the name of the identifier property.
+Its bridge will be the identifier bridge referenced by the embedded type's
+<<mapping-identifiermapping-explicit,`@DocumentId` annotation>>, if any,
+or the default value bridge for the identifier property type's, by default.
+
+[TIP]
+====
+If you need more advanced mapping (custom name, custom bridge, sortable, ...),
+do not use `includeEmbeddedObjectId`.
+
+Instead, define the field explicitly in the indexed-embedded type
+by annotating the identifier property with <<mapping-directfieldmapping,`@GenericField` or a similar field annotation>>,
+and make sure the field is included by `@IndexedEmbedded`
+by <<mapper-orm-indexedembedded-filtering,configuring filters as necessary>>.
```

Comment:
just wondering, would filters work on implicit ids too?

Reply:
They should, yes, as they are handled at a rather low level, essentially as callbacks called by the backend.